File: Models/AcompanhamentoPedido.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace pizzaria;

public class AcompanhamentoPedido
{
    [Key]
    public int Id { get; set; }
    public Acompanhamento Acompanhamento { get; set; }
    public int Quantidade { get; set; }
    public double PrecoTotal { get; private set; }
    [JsonIgnore]
    public PedidoFinal? PedidoFinal { get; set; }

    public AcompanhamentoPedido() { }

    public AcompanhamentoPedido(int id, Acompanhamento acompanhamento, int quantidade)
    {
        Id = id;
        Acompanhamento = acompanhamento;
        Quantidade = quantidade;
        CalcularPreco();
    }

    public void CalcularPreco()
    {
        PrecoTotal = Acompanhamento.Preco * Quantidade;
    }

    public override string ToString()
    {
        Console.Write($"Acompanhamento: {Acompanhamento.Nome} | ");
        Console.Write($"Preço Unitário: R${Acompanhamento.Preco} | ");
        Console.Write($"Quantidade: {Quantidade} | ");
        Console.Write($"Preço Total do Acompanhamento: R${PrecoTotal}");
        return "";
    }
}

File: Frontend/src/app/models/AcompanhamentoPedido.ts
import { Acompanhamento } from "./Acompanhamento";
import {Pedido} from "./Pedido";

export class AcompanhamentoPedido implements Pedido {
  id: number = 0;
  acompanhamento: Acompanhamento = new Acompanhamento();
  quantidade: number = 0;
  preco: number = 0;

  constructor(acompanhamento: Acompanhamento, quantidade: number) {
    this.acompanhamento = acompanhamento;
    this.quantidade = quantidade;
    this.calcularPreco();
  }

  calcularPreco() {
    this.preco = this.acompanhamento.preco * this.quantidade;
  }

  getDescricao(): string {
    return this.acompanhamento.nome;
  }

  getPreco(): number {
    return this.preco;
  }

  getQuantidade(): number {
    return this.quantidade;
  }

}
